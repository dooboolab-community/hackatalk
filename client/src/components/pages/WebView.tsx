import React, {FC} from 'react';
import {RouteProp, useRoute} from '@react-navigation/core';

import {RootStackParamList} from '../navigations/RootStackNavigator';
import {WebView} from 'react-native-webview';
import styled from 'styled-components/native';

const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${({theme}) => theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const Page: FC = () => {
  const {
    params: {uri},
  } = useRoute<RouteProp<RootStackParamList, 'WebView'>>();

  return (
    <Container>
      <WebView source={{uri}} />
    </Container>
  );
};

export default Page;
