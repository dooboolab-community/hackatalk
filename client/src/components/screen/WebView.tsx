import React, { ReactElement } from 'react';
import { RootStackNavigationProps, RootStackParamList } from '../navigation/RootStackNavigator';

import { RouteProp } from '@react-navigation/core';
import { WebView } from 'react-native-webview';
import styled from 'styled-components/native';

const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'WebView'>;
  route: RouteProp<RootStackParamList, 'WebView'>;
}

function Page(props: Props): ReactElement {
  const { route: { params: { uri } } } = props;

  return (
    <Container>
      <WebView source={{ uri }} />
    </Container>
  );
}

export default Page;
