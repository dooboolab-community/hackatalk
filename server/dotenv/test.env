JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
