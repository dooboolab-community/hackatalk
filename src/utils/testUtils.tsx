import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
