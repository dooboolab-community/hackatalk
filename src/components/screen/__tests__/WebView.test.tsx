import 'react-native';

import * as React from 'react';

import {
  RenderResult,
  render,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import WebView from '../WebView';

// eslint-disable-next-line
let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

describe('[WebView] screen', () => {
  beforeEach(() => {
    props = createTestProps({
      route: {
        params: {
          uri: 'https://dooboolab.com',
        },
      },
    });
    component = createTestElement(<WebView {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    expect(testingLib.baseElement).toMatchSnapshot();
  });
});
