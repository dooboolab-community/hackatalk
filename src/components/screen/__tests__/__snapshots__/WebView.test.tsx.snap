// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[WebView] screen renders without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <RNCWebView
          cacheEnabled={true}
          javaScriptEnabled={true}
          messagingEnabled={false}
          source={
            Object {
              "uri": "https://dooboolab.com",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
            ]
          }
          useSharedProcessPool={true}
        />
      </View>
    </SafeAreaView>
  </View>
</View>
`;
